src/Executor/Setup.cs

```diff
@@ -0,0 +1,290 @@
+﻿using System;
+using System.IO;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Microsoft.DotNet.Execute
+{
+    public class Setup
+    {
+        public Dictionary<string, Setting> Settings { get; set; }
+        public Dictionary<string, Command> Commands { get; set; }
+        public Dictionary<string, Tool> Tools { get; set; }
+        public Dictionary<string, string> SettingParameters { get; set; }
+        public string Os { get; set; }
+        public string ConfigurationFilePath { get; set; }
+
+
+        private string FindSettingValue(string valueToFind)
+        {
+            if (Settings.ContainsKey(valueToFind))
```

Comment:
This double access is better handled by Settings.TryGetValue(valueToFind, out foundValue)